JsonPatch.Tests/ExtensionTests.cs

```diff
@@ -0,0 +1,212 @@
+			Assert.AreEqual(op.Operations.Count, 1);
+			Assert.AreEqual(op.Operations[0].Op, OperationType.Remove);
+
+			Assert.AreNotEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+		}
+
+		[Test]
+		public void ReplaceArray_Test()
+		{
+			var e1 = JsonDocument.Parse("[1,2,3]");
+			var e2 = JsonDocument.Parse("[1,2,1]");
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			Assert.AreEqual(op.Operations.Count, 1);
+			Assert.AreEqual(op.Operations[0].Op, OperationType.Replace);
+
+			Assert.AreNotEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1.RootElement.ToString(),e2.RootElement.ToString());
+		}
+
+
+		[Test]
+		public void ComplexObject_Test()
+		{
+			var e1 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,2,3]},{\"test\":[1,2,3]}]"),
+				Name = "Test",
+				Numbers = new[] {1, 2, 3},
+				Strings = new[] {"test1", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.NewGuid(), Name = "TestNameInner1", Numbers = new[] {3, 2, 1}, Strings = new[] {"Test3", "test4"}}}
+			};
+			var e1Json = JsonSerializer.Serialize(e1);
+			var e2 = new TestModel()
+			{
+				Id = Guid.NewGuid(),
+				Attributes = JsonDocument.Parse("[{\"test1\":\"test123\"},{\"test\":\"test32132\"},{\"test1\":\"test321\"},{\"test\":[1,1,3]}]"),
+				Name = "Test4",
+				Numbers = new[] {1, 2, 3, 4},
+				Strings = new[] {"test2", "test2"},
+				InnerObjects = new List<TestModel>() {new TestModel() {Id = Guid.NewGuid(), Name = "TestNameInner1", Numbers = new[] {1, 2, 1}, Strings = new[] {"Test3", "test4", "test5"}}}
+			};
+			var e2Json = JsonSerializer.Serialize(e2);
+			var op = e1.CreatePatch(e2, new JsonSerializerOptions() {IgnoreNullValues = true});
+			var opBack = e2.CreatePatch(e1, new JsonSerializerOptions() {IgnoreNullValues = true});
+
+			Assert.AreNotEqual(e1, e2);
+			e1 = op.ApplyPatch(e1);
+			Assert.AreEqual(e1, e2);
+			e1 = opBack.ApplyPatch(e1);
+			Assert.AreNotEqual(e1, e2);
+			Assert.AreEqual(JsonSerializer.Serialize(e1), e1Json);
+		}
+	}
+
+	public static class Extensions
+	{
+		public static T ApplyPatch<T>(this Json.Patch.JsonPatch patch, T obj)
+		{
+			using var doc = JsonDocument.Parse(JsonSerializer.Serialize(obj));
+			var res = patch.Apply(doc.RootElement).Result;
+			var result = JsonSerializer.Deserialize<T>(res.GetRawText());
+			return result;
+		}
+	}
+
+	public class TestModel
+	{
+		public Guid Id { get; set; }
+		public string Name { get; set; }
+		public int[] Numbers { get; set; }
+		public string[] Strings { get; set; }
+		public List<TestModel> InnerObjects { get; set; }
+		public JsonDocument Attributes { get; set; }
```

Comment:
I'm not sure what including this property is accomplishing.  It seems to be making the tests a bit confusing.

Reply:
We use JsonDocument in our projects when we need some dynamic objects here. So I thought that it would be quite right to test this too.

Reply:
`JsonDocument` is tested implicitly becuase you convert objects to `JsonDocument` before generating the patch.  I don't think this is explicitly necessary in the tests.